### DISCUSSION TITLE
Deriving ECGs

### DISCUSSION THREAD

**Discussion by edorachlee** [rsachetto/MonoAlg3D_C#74] [General]
Here goes another discussion.

The built-in ECG derivation function is super useful. However I am not clear on how the reference potential, i.e. ground, is set while calculating $$\Phi_e$$. How is the reference point decided? Here I am assuming pseudo-bidomain.

Reading through [a pseudo-bidomain paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3378475) it seems the authors chose a reference point within an infinite bath in Fig 1, though it is not shown exactly how they determined the location of the reference point.

I have been looking at [CALC_ECG](https://github.com/rsachetto/MonoAlg3D_C/blob/a7fc88a36dc27262b67eee97a878337416065896/src/ecg_library/ecg.c#L241) and [calc_reference_potential](https://github.com/rsachetto/MonoAlg3D_C/blob/a7fc88a36dc27262b67eee97a878337416065896/scripts/calc_apd_specific_cells/calc_apd_specific_cells.py#L81) but am unfortunately still unclear. 

I am asking because it is unclear to me whether I need to include a ground electrode(typically right leg, aka RL), and account for the reference point that way, or if it is already accounted for while solving pseudo-bidomain.

Best,
Lucas

**Comment by rsachetto** [comment]
> Here goes another discussion.
> 
> The built-in ECG derivation function is super useful. However I am not clear on how the reference potential, i.e. ground, is set while calculating Φ e . How is the reference point decided? Here I am assuming pseudo-bidomain.

Yes, it is pseudo-bidomain.

Although I've implemented the function. I'am not very familiarized how to use it. I know that there is an additional step that is needed to generate the 12-lead ECG. This additional step is not included in the simulator. The output of the ecg function is used to generate the final result. 

Basically, the simulator solves equation 11 of the pseudo-bidomain paper for each lead.

**Reply by edorachlee** [reply]
So for clarity I am just reproducing Equation 11 here: $$\Phi_e=\frac{1}{4\pi\sigma_b} \int_\Omega \frac{\beta I_m}{||\textbf{r}||}d\Omega$$

When calculaiting the distance vector $$||\textbf{r}||$$, how are the source and field points derived?

**Comment by rsachetto** [comment]
Talking to a a colleague that uses the function, he sent me a snippet of code that he uses to generate a 8 lead ECG:

```python
def import_simulated_ecg_8leads_raw(filename):

    data = np.genfromtxt(filename, skip_footer=1)
    t = data[:, 0]
    LA = data[:, 1]
    RA = data[:, 2]
    LL = data[:, 3]
    RL = data[:, 4]
    V1 = data[:, 5]
    V2 = data[:, 6]
    V3 = data[:, 7]
    V4 = data[:, 8]
    V5 = data[:, 9]
    V6 = data[:, 10]

    # Ealuate Wilson's central terminal
    VW = 1.0 / 3.0 * (RA + LA + LL)

    # Evaluate simulated ECG lead traces
    V1 = V1 - VW
    V2 = V2 - VW
    V3 = V3 - VW
    V4 = V4 - VW
    V5 = V5 - VW
    V6 = V6 - VW
    I = LA - RA
    II = LL - RA
    III = LL - LA
    aVL = LA - (RA + LL) / 2.0
    aVF = LL - (LA + RA) / 2.0
    aVR = RA - (LA + LL) / 2.0
    ecgs = np.vstack((I, II, V1, V2, V3, V4, V5, V6))
```

**Reply by edorachlee** [reply]
Ah, this is very similar to my own setup. So from his code it looks like he is not using RL(though recorded) to calculate his ecgs. This helps clear up my second question. Thank you.
